django/db/models/query_utils.py

```diff
@@ -175,6 +175,26 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(self.identity)
 
+    @cached_property
+    def referenced_base_fields(self):
+        """
+        Retrieve all base fields referenced directly or through F expressions excluding
+        any fields referenced through joins.
```

Comment:
This docstring should be wrapped at 79 cols, so:
```suggestion
        Retrieve all base fields referenced directly or through F expressions
        excluding any fields referenced through joins.
```
Also, question: would there be a way to add a specific case in the added test to make it explicit how "any fields referenced through joins" are excluded?